src/main/java/seedu/address/storage/JsonAdaptedApplication.java

```diff
@@ -140,7 +145,16 @@ public Application toModelType() throws IllegalValueException {
         }
         final JobTitle modelJobTitle = new JobTitle(jobTitle);
 
+        if (details == null) {
```

Comment:
I think this can be improved to use the `requireNonNull()` method as part of defensive programming.

Reply:
Agreed, will standardize all boolean tests using these methods moving forward